### PR TITLE
Bump and pin dependencies for dev preview

### PR DESCRIPTION
**Pull request by gibizer**

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -9,10 +9,10 @@ require (
 	github.com/openshift/api v3.9.0+incompatible
 	github.com/openstack-k8s-operators/infra-operator/apis v0.0.0-20230628130307-16734cb02944
```
Comment (gibizer) [inline go.mod:10]: We need to wait for the infra-operator/api to be tagged

**Review by stuggi — APPROVED** [review]
/lgtm

**Pull request merged**